cluster/assemblestate/devices.go

```diff
@@ -0,0 +1,256 @@
+package assemblestate
+
+import (
+	"time"
+)
+
+// DeviceQueryTracker manages device identity information and query/response
+// lifecycle for unknown devices in the cluster.
+type DeviceQueryTracker struct {
+	// responses is used to send notifications that indicate that we are able to
+	// respond to a peer's query for device information.
+	responses chan struct{}
+
+	// queries is used to send notifications that indicate that we should query
+	// a peer for device information.
+	queries chan struct{}
+
+	// inflight keeps track of queries for device information that we have
+	// in-flight. We keep track of a mapping of device RDT to time at which the
+	// query was sent.
+	inflight map[DeviceToken]time.Time
+
+	// timeout is the amount of time we will wait before sending a query for
+	// device information again.
+	timeout time.Duration
+
+	// unknowns keeps track of which devices each peer has queried us for.
```

Comment:
peer is the first level of the maps? but is not immediately clear as the order in the sentence is the reverse, this need somehow to expanded to clarify that